doc/radosgw/nfs.rst

```diff
@@ -0,0 +1,337 @@
+an attribute-setting operation such as chown or chmod) always have a
+leaf object representation used to store materialized attributes such
+as Unix ownership and permissions.
+
+Supported Operations
+====================
+
+The RGW NFS interface supports most operations on files and
+directories, with the following restrictions:
+
+- Links, including symlinks, are not supported
+- Directories may not be moved/renamed
+
+  + files may be moved between directories
+
+- Only full, sequential *write* i/o is supported
+
+  + i.e., write operations are constrained to be **uploads**
+  + many typical i/o operations such as editing files in place will necessarily fail as they perform non-sequential stores
+  + some file utilities *apparently* writing sequentially (e.g., some versions of GNU tar) may fail due to infrequent non-sequential stores
+  + When mounting via NFS, sequential application i/o can generally be constrained to be written sequentially to the NFS server via a synchronous mount option (e.g. -osync in Linux)
+  + NFS clients which cannot mount synchronously (e.g., MS Windows) will not be able to upload files
+
+Security
+========
+
+The RGW NFS interface provides a hybrid security model with the
+following characteristics:
+
+- NFS protocol security is provided by the NFS-Ganesha server, as negotiated by the NFS server and clients
+
+  + e.g., clients can by trusted (AUTH_SYS), or required to present Kerberos user credentials (RPCSEC_GSS)
+  + RPCSEC_GSS wire security can be integrity only (krb5i) or integrity and privacy (encryption, krb5p)
+  + various NFS-specific security and permission rules are available
+
+    * e.g., root-squashing
+
+- a set of RGW/S3 security credentials (unknown to NFS) is associated with each RGW NFS mount (i.e., NFS-Ganesha EXPORT)
+
+  + all RGW object operations performed via the NFS server will be performed by the RGW user associated with the credentials stored in the export being accessed (currently only RGW and RGW LDAP credentials are supported)
+
+    * additional RGW authentication types such as Keystone are not currently supported
+
+Configuring an NFS-Ganesha Instance
+===================================
+
+Each NFS RGW instance is an NFS-Ganesha server instance *embeddding*
+a full Ceph RGW instance.
+
+Therefore, the RGW NFS configuration includes Ceph and Ceph Object
+Gateway-specific configuration in a local ceph.conf, as well as
+NFS-Ganesha-specific configuration in the NFS-Ganesha config file,
+ganesha.conf.
+
+ceph.conf
+---------
+
+Required ceph.conf configuration for RGW NFS includes:
+
+* valid [client.radosgw.{instance-name}] section
+* valid values for minimal instance configation, in particular, an installed and correct *keyring*
+
+Other config variables are optional, front-end-specific and front-end
+selection variables (e.g., *rgw data* and *rgw frontends*) are
+optional and in some cases ignored.
+
+A small number of config variables (e.g., *rgw_namespace_expire_secs*)
```

Comment:
other radosgw docs use the ` ``double-backtick`` ` syntax to highlight config variables, config sections, and other terms. see https://raw.githubusercontent.com/ceph/ceph/master/doc/radosgw/config-ref.rst for examples